avocado/utils/memory.py

```diff
@@ -41,6 +108,20 @@ def read_from_meminfo(key):
     return int(re.search(r'\d+', meminfo).group(0))
```

Comment:
This is not related to your commit, but your commit made me read this function, so here we go.
IMO, this function is should be improved. It's using `grep` and a regex search, and both are not really needed.
I'd suggest to parse the `/proc/meminfo` into a dictionary and, accepting `None` key, return the whole dictionary. If some key is provided, than only that key value is returned. It will increase the flexibility and the usability.
Something like:

```python
def read_from_meminfo(key=None):
    meminfo = {}
    with open('/proc/meminfo', 'r') as f:
        for line in f.readlines():
            item, value = line.strip().split(None, 1)
            meminfo[item.strip(':')] = value.strip(' kB')
    if key is not None:
        return meminfo[key]
    return meminfo
```

Usage would be:
```python

>>> read_from_meminfo()
{'WritebackTmp': '0', 'SwapTotal': '7930876', 'Active(anon)': '937468', 'SwapFree': '7930876', 'DirectMap4k': '374784', 'KernelStack': '7312', 'MemFree': '9732968', 'HugePages_Rsvd': '0', 'Committed_AS': '5107800', 'Active(file)': '1240388', 'NFS_Unstable': '0', 'VmallocChunk': '0', 'CmaFree': '0', 'Writeback': '0', 'Inactive(file)': '1698816', 'MemTotal': '15853856', 'VmallocUsed': '0', 'DirectMap1G': '6291456', 'HugePages_Free': '0', 'AnonHugePages': '0', 'AnonPages': '1748280', 'Active': '2177856', 'Inactive(anon)': '940808', 'CommitLimit': '15857804', 'Hugepagesize': '2048', 'Cached': '3067080', 'SwapCached': '0', 'VmallocTotal': '34359738367', 'CmaTotal': '0', 'Shmem': '130140', 'Mapped': '375208', 'ShmemPmdMapped': '0', 'SUnreclaim': '198616', 'Unevictable': '32', 'SReclaimable': '535092', 'MemAvailable': '12868908', 'Mlocked': '32', 'DirectMap2M': '10575872', 'HugePages_Surp': '0', 'Bounce': '0', 'Inactive': '2639624', 'PageTables': '26068', 'HardwareCorrupted': '0', 'ShmemHugePages': '0', 'HugePages_Total': '0', 'Slab': '733708', 'Buffers': '2264', 'Dirty': '1736'}

>>> read_from_meminfo('Active')
'2179888'

>>> read_from_meminfo()['Slab']
'733708'

>>> read_from_meminfo('foo')
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "<stdin>", line 9, in read_from_meminfo
KeyError: 'foo'
```